### PR TITLE
Add some tests for Personal Info section

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
This adds some tests for the Personal Info section:
- personal info shows up
- contact info shows up
- can add email address (five tests to handle different ways the process can succeed or fail)

This also converts some `*ByRole` queries to `*ByText` where appropriate. I noticed in using the `ByRole` queries seem to take 100x (!!) as much time (~4ms to >400ms) to resolve. So from this point forward I'll use those queries judiciously.

## Testing done
Local + all the new tests

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/user/profile/vet360/components/ContactInfoForm.jsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import SchemaForm from 'platform/forms-system/src/js/components/SchemaForm';
 
 const ContactInfoForm = props => (
   <SchemaForm
+    addNameAttribute
```
Comment (erikphansen) [inline src/platform/user/profile/vet360/components/ContactInfoForm.jsx:8]: This is needed so that we can query the form with `getByRole('form')`

**Comment by erikphansen** [comment]
@sanlouise sorry! It's 900 lines but so much of that is just mock API responses and mock Redux data! It's "only" 400+ lines of real tests. And a lot of that is duplicative code. I'm taking another pass to see what else can be cleaned up, but feel free to start looking this over.

File: src/applications/personalization/profile-2/msw-mocks.js [hunk 2]
```diff
@@ -60,3 +60,353 @@ export const updateDirectDepositFailure = [
...
+                '21P-530',
+                '21P-527EZ',
+                '686C-674',
+                '20-0996',
+                'MDOT',
+              ],
+              vet360ContactInformation: {
+                email: {
+                  createdAt: '2020-07-28T14:57:54.000Z',
+                  emailAddress: newEmailAddress,
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/msw-mocks.js:270]: note that we are returning the `newEmailAddress` that was part of the POST call...

File: src/applications/personalization/profile-2/msw-mocks.js [hunk 3]
```diff
@@ -60,3 +60,353 @@ export const updateDirectDepositFailure = [
...
+    return res(
+      ctx.status(400),
+      ctx.json({
+        // This is an error response I saw when trying to start a phone number
+        // update, not email address update. I have not seen an error when
+        // trying to create an email address update transaction so am using this
+        // for the mock response
+        errors: [
+          {
+            title: 'Check Domestic Phone Number',
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/msw-mocks.js:76]: Could we use an email error response from `config/locales/exceptions.en.yml` in `vets-api`?

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.add-email-address.unit.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+  // before we can use msw, we need to make sure that global.fetch has been
+  // restored and is no longer a sinon stub.
+  let server;
+  before(() => {
+    resetFetch();
+    server = setupServer(...mocks.addEmailAddressSuccess());
+    server.listen();
+  });
+  beforeEach(() => {
+    window.VetsGov = { pollTimeout: 1 };
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.add-email-address.unit.spec.jsx:53]: What does this `pollTimeout` do?

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.unit.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    },
+    user: {
+      profile: {
+        vet360: getBasicContactInfoState(),
+      },
+    },
+  };
+}
+
+function fillOutAndSubmitBankInfoForm(view) {
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.unit.spec.jsx:42]: Is this logic related to Direct Deposit supposed to live in a separate test file and actually being called there?

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.add-email-address.unit.spec.jsx (hunk 4 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.add-email-address.unit.spec.jsx:53]: This is a convention that's been used in the past to determine what to set timeout/interval delays as so we can use a much shorter delay when running unit tests. In this particular case this shortens the polling delay when we check on the status of a pending transaction. Under normal situations we check on the status of the transaction every 1000ms, but in unit tests we'll check every 1ms. I'm also using the existence of this global flag to determine if our redux actions creators should make real API calls (in `isVet360Configured()`) and changing how aggressively we automatically exit edit mode when a save is taking a long time (ie, we change the definition of what a "long time" means in the context of waiting for a profile save to happen from 5000ms to 50ms).

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.unit.spec.jsx (hunk 5 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.unit.spec.jsx:42]: This was my being sloppy :) Thanks

**Review by sanlouise — APPROVED** [review]

File: src/applications/personalization/profile-2/msw-mocks.js (hunk 3 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/msw-mocks.js:76]: whoa... I did not know this file existed

**Pull request merged**